CMakeLists.txt

```diff
@@ -12,7 +12,7 @@ if(WIN32)
   cmake_policy(SET CMP0091 OLD)
   set(CMAKE_SKIP_TEST_ALL_DEPENDENCY TRUE)
 endif()
-if(APPLE AND (NOT CMAKE_OSX_SYSROOT OR CMAKE_OSX_SYSROOT STREQUAL ""))
+if((CMAKE_GENERATOR MATCHES Xcode) AND (NOT CMAKE_OSX_SYSROOT OR CMAKE_OSX_SYSROOT STREQUAL ""))
```

Comment:
Is it really dependent of the Generator or is it the compiler (XCode's clang)?  (eg I use Ninja+XCode's clang).

Reply:
I don't know, but you have the perfect setup to figure this out then! Does this PR build for you? If not, it's the compiler and I need to put a check for Apple Clang there. Can you give a hand by doing this check?

Reply:
will do ...

Reply:
So form a fresh directory, with your commit, it seems to work ... but then again it is still 'set':
```
mac-142865:osxroot pcanal$ rg -C3 CMAKE_OSX_SYSROOT CMakeCache.txt 
475-
476-//The product will be built against the headers and libraries located
477-// inside the indicated SDK.
478:CMAKE_OSX_SYSROOT:PATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.5.sdk
479-
```

Reply:
Yes, I am fine with this ... however I can't help to ponder if CMake found a 'better' solution since they no longer use it (seemingly)

Reply:
Apparently it is still needed by 'just' for the case when using XCode's compiler ... oh well the change here might be good enough (the risk is that the SDK found conflicts with the one from the compiler when it comes from homebrew for example ... )